aicsimageio/aics_image.py

```diff
@@ -131,6 +131,72 @@ def data(self):
                                                  return_dims=self.dims)
         return self._data
 
+    def size(self, dims: str = "STCZYX"):
+        """
+        Parameters
+        ----------
+        dims: A string containing a list of dimensions being requested.  The default is to return the 6 standard dims
+
+        Returns
+        -------
+        Returns a tuple with the requested dimensions filled in
+        """
+        return tuple([self.data.shape[self.dims.index(dim)] for dim in dims])
+
+    @property
+    def size_x(self):
+        """
+        Returns
+        -------
+        Returns the x size
+        """
+        return self.size("X")[0]
+
+    @property
+    def size_y(self):
+        """
+        Returns
+        -------
+        Returns the y size
+        """
+        return self.size("Y")[0]
+
+    @property
+    def size_z(self):
+        """
+        Returns
+        -------
+        Returns the z size
+        """
+        return self.size("Z")[0]
+
+    @property
+    def size_c(self):
+        """
+        Returns
+        -------
+        Returns the c size
+        """
+        return self.size("C")[0]
+
+    @property
+    def size_t(self):
+        """
+        Returns
+        -------
+        Returns the t size
+        """
+        return self.size("T")[0]
+
+    @property
+    def size_s(self):
+        """
+        Returns
+        -------
+        Returns the s size
+        """
+        return self.size("S")[0]
```

Comment:
I like the functionality but might I suggest 
``` python
    def __init__(self):
        self.pattern = re.compile(r'size_([A-z])$')

    def __getattr__(self, attr_name):
        match = re.match(self.pattern, attr_name)
        if match:
            return self.size(match.group(1))[0]
        else:
            raise AttributeError(f'No such attribute: {attr_name}')

    def size(self, dims: str = "STCZYX"):
        upper_dims = dims.upper()
        if len(dims) != len(set(dims).intersection(set(self.dims))):
            print("Dimension not present in data!")
            return tuple([0])
        return tuple([self.data.shape[self.dims.index(dim)] for dim in upper_dims])
```

Reply:
This kind of thing is popular in ruby but I'd not sure how it's thought of in python

Reply:
I will definitely take your cue about making sure to convert to upper, and assert that the requested dims are valid.  But I'm not sure I understand the ruby comment.  What is the thing that's popular in ruby?